### ISSUE TITLE
About Debug features.

### ISSUE THREAD

**Issue by floydGG** [nipkownix/re4_tweaks#169]
First of all thank you for your awesome tweaks, i finally can play this game with keyboard and mouse (before i found your tweaks, it is horrible aiming with mouse).
And second thing about debug, i really love the disable fog or others visual effect, i'm playing on old laptop with low specs.
I don't know if it's possible or not, but can u make the removing fog,shadows,or any other visual effect including in the menu without activating the debug options?

I'm sorry if this is right place to ask something like this, but i don't know how/who contact about this awesome tweaks.

**Comment by emoose** [comment]
Should be pretty easy to do that with Cheat Engine, `bio4.exe+0x85A7B8` is the address for DISP flags in 1.1.0, if you check that with the fog/shadow options enabled/disabled you should be able to see what bit is changed for it & change it manually.

Not sure about about adding trainer stuff like that to re4_tweaks, it's pretty simple to do it with tools like CE/Ultimate trainer already.

**Comment by Mister-Curious** [comment]
@emoose are you guys interested in maybe making some custom edits to restore some functionality here? Im fairly sure my associate could create some code to remove enemies, and other things that could be patched into the menus.

**Comment by floydGG** [comment]
> Should be pretty easy to do that with Cheat Engine, `bio4.exe+0x85A7B8` & `bio4.exe+0x85A7BC` are the addresses for DISP flags in 1.1.0, if you check that with the fog/shadow options enabled/disabled you should be able to see what bit is changed for it & change it manually.
> 
> Not sure about about adding trainer stuff like that to re4_tweaks, it's pretty simple to do it with tools like CE/Ultimate trainer already, but who knows, maybe in future a flag editor could be added to the UI so we aren't stuck with that keyboard-only debug one.

But i'm using the 1.0.6 still, i do use raz0r trainer but i don't see features to turn off the fog thing.
It's the addresses works too for 1.0.6 ? 
Anyway thank you for the explanation about it, i might try with cheat engine

**Comment by emoose** [comment]
> @emoose are you guys interested in maybe making some custom edits to restore some functionality here? Im fairly sure my associate could create some code to remove enemies, or disable collision and other things that could be patched into the debug menus. While its not an 'authentic' fix, it would still add functionality.

Yeah I was thinking of looking into that some time, did something similar with NieR Automata copying flag checks from the debug ver to the release, unfortunately the RE4 PC debug ver doesn't seem to be a full debug version though (missing a lot compared to the GC debug at least..), can probably figure out things from that GC build though.

Any additions from you or your friend would be welcomed, doesn't have to match with what GC debug does exactly, but restoring at least some of the flags functionality would be nice.

Haven't really gone through the list at all myself, maybe something to look into if anyone gets bored.

> But i'm using the 1.0.6 still, i do use raz0r trainer but i don't see features to turn off the fog thing. It's the addresses works too for 1.0.6 ? Anyway thank you for the explanation about it, i might try with cheat engine

Nah 1.0.6 would use different addrs, had a quick look and think it should be `bio4.exe+0x856F38` & `bio4.exe+0x856F3C`

**Comment by floydGG** [comment]
> 
> > But i'm using the 1.0.6 still, i do use raz0r trainer but i don't see features to turn off the fog thing. It's the addresses works too for 1.0.6 ? Anyway thank you for the explanation about it, i might try with cheat engine
> 
> Nah 1.0.6 would use different addrs, had a quick look and think it should be `bio4.exe+0x856F38` & `bio4.exe+0x856F3C`, for the uncensored 1.0.6 at least. If nothing there seems to help, maybe try looking at the bytes from `bio4.exe+0x856F40` through `bio4.exe+857050`, those ones should control the DBG_* flags anyway.

Thank you so much, gotta try this.

**Comment by Mister-Curious** [comment]
@emoose So my engineer buddy and I poked around a little bit yesterday in the debug. He was not able to see any activity in Cheat Engine while toggling various flags, so at this point I am wondering if you are interested in having a conversation with us on Discord on how to proceed. I see many functions in the flag page I know he could patch so long as we figured out how to hook into the debug flags.  

Please, if you have time, email me and we can set up a time to have a live chat which I think is the best way forward. Either that or just hop onto Discord and message me (i have left several messages for you there with no replies). 

Cheers

**Comment by psycowrld** [comment]
@floydGG can you make a video tutorial bro for the cheat engine things i want to remove shadow blur and filter for performance too.
![1661878981580](https://user-images.githubusercontent.com/113557334/190263746-850b9b19-2626-4a88-8d10-c0f457041e77.png)

**Comment by brad86m** [comment]
Is there a way to reset an event flag in a room ?
I stupidly used the save game within debug menu whilst fighting Saddler at the end of the game. Upon loading, I get the stage, but with nothing else. No Saddler, just the room that I can run around in. The levers do not work, the bridges do not go up and down. Nothing!. I have tried area jumping to try and trigger the fight again, but nothing happens. Jumping directly to the ending from here results in the island blowing up cutscene, and with Leon dying. I can't trigger the speed boat segment to be able to finish. I wanted to keep my loadout for a NG+ playthrough, so any way to fix the end would be nice.

**Comment by emoose** [comment]
@brad86m might have found how to change those flags, I've made a test build here: 
[re4_tweaks-roomSaveTest.zip](https://github.com/nipkownix/re4_tweaks/files/10230488/re4_tweaks-roomSaveTest.zip)

If you load into the save and then open re4_tweaks menu (F1) -> Trainer -> Flag Editor -> choose "ROOM_SAVE" from the dropdown, that should give you a list of flags that are set
You could then try unsetting them, then use Trainer -> Save Game to save the game, and hopefully if you reload the save it should have that flag applied (or unapplied)
